query-planner-js/src/__tests__/buildPlan.defer.test.ts

```diff
@@ -0,0 +1,3090 @@
+import { operationFromDocument, Schema, ServiceDefinition } from '@apollo/federation-internals';
+import gql from 'graphql-tag';
+import { QueryPlanner } from '@apollo/query-planner';
+import { composeAndCreatePlanner, composeAndCreatePlannerWithOptions } from "./buildPlan.test";
+
+function composeAndCreatePlannerWithDefer(...services: ServiceDefinition[]): [Schema, QueryPlanner] {
+  return composeAndCreatePlannerWithOptions(services, { deferStreamSupport: { enableDefer : true }});
+}
+
```

Comment:
Great test suite.

One edge case I don't see tests for is when a single field is selected in the same selection set both deferred and not deferred (which in practice is likely to occur due to named fragment application) apparently the spec currently expects you to send the repeated field both in the initial response and a deferred response.

Reply:
fixed in #2093, thanks